typescript/phoenix-sdk/src/client.ts

```diff
@@ -50,75 +110,326 @@ export class Client {
+    if (buffer === undefined) {
+      throw new Error("Unable to get market account data");
+    }
+
     const market = await Market.load({
-      connection: this.connection,
-      address: new PublicKey(marketAddress),
+      address: marketKey,
+      buffer,
+      tokenList: this.tokenConfig,
     });
-    this.markets.push(market);
     for (const token of [market.baseToken, market.quoteToken]) {
       const mint = token.data.mintKey.toBase58();
       if (this.tokens.find((t) => t.data.mintKey.toBase58() === mint)) continue;
       this.tokens.push(token);
     }
+    this.markets.set(marketAddress, market);
+    const clockBuffer = accounts[1]?.data;
+    if (clockBuffer === undefined) {
+      throw new Error("Unable to get clock");
+    }
+    this.reloadClockFromBuffer(clockBuffer);
+  }
+
+  /**
+   * Refreshes the market data and clock
+   *
+   * @param marketAddress The address of the market to refresh
+   *
+   * @returns The refreshed Market
+   */
+  public async refreshMarket(
+    marketAddress: string | PublicKey
+  ): Promise<Market> {
+    const marketKey = new PublicKey(marketAddress);
+    const existingMarket = this.markets.get(marketKey.toString());
+    if (existingMarket === undefined) {
+      throw new Error("Market does not exist: " + marketKey.toBase58());
+    }
+    const accounts = await this.connection.getMultipleAccountsInfo(
+      [marketKey, SYSVAR_CLOCK_PUBKEY],
+      "confirmed"
+    );
+    if (accounts.length !== 2)
+      throw new Error("Account not found for market: " + marketKey.toBase58());
+
+    const buffer = accounts[0]?.data;
+    if (buffer === undefined) {
+      throw new Error("Unable to get market account data");
+    }
+    existingMarket.reload(buffer);
+    const clockBuffer = accounts[1]?.data;
+    if (clockBuffer === undefined) {
+      throw new Error("Unable to get clock");
+    }
+    this.reloadClockFromBuffer(clockBuffer);
+
+    return existingMarket;
+  }
+
+  public async reloadClock() {
+    const clockAccount = await this.connection.getAccountInfo(
+      SYSVAR_CLOCK_PUBKEY,
+      "confirmed"
+    );
+    const clockBuffer = clockAccount?.data;
+    if (clockBuffer === undefined) {
+      throw new Error("Unable to get clock");
+    }
+
+    this.reloadClockFromBuffer(clockBuffer);
+  }
+
+  reloadClockFromBuffer(clockBuffer: Buffer) {
+    this.clock = deserializeClockData(clockBuffer);
+  }
+
+  /**
+   * Returns the market's ladder of bids and asks
+   * @param marketAddress The `PublicKey` of the market account
+   */
+  public getLadder(
+    marketAddress: string,
+    levels: number = DEFAULT_LADDER_DEPTH
+  ): Ladder {
+    const market = this.markets.get(marketAddress);
+    if (!market) throw new Error("Market not found: " + marketAddress);
+    return getMarketLadder(
+      market.data,
+      this.clock.slot,
+      this.clock.unixTimestamp,
+      levels
+    );
+  }
+
+  /**
+   * Returns the market's ladder of bids and asks  as JS numbers
```

Comment:
extra space in doc. also, this doc doesn't really explain the difference between this method and above.